test/performance/alignment/edit_distance_benchmark_template.hpp

```diff
@@ -0,0 +1,98 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#pragma once
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/test/alignment/align_pairwise_edit_distance.hpp>
+#include <seqan3/test/performance/units.hpp>
+#include <seqan3/test/performance/sequence_generator.hpp>
+#include <seqan3/test/seqan2.hpp>
+
+// ----------------------------------------------------------------------------
+//  seqan3 edit distance pairwise alignment benchmarks
+// ----------------------------------------------------------------------------
+
+//!\brief This directly benchmarks the alignment algorithm without going through the whole align_pairwise function
+template <typename sequence_pair_generator_t, typename align_configs_t>
+void seqan3_align_pairwise_edit_distance_benchmark(benchmark::State & state,
+                                                   sequence_pair_generator_t sequence_pair_generator,
+                                                   align_configs_t && edit_distance_cfg)
+{
+    using seqan3::test::edit_distance_algorithm;
+
+    auto sequence_pair_or_pairs = sequence_pair_generator(state);
+    constexpr bool is_collection = sequence_pair_generator.is_collection;
+    using sequence_t = typename sequence_pair_generator_t::sequence_t;
+
+    int64_t total = 0;
+
+    auto algorithm = edit_distance_algorithm::select<sequence_t, sequence_t>(edit_distance_cfg);
+    for (auto _ : state)
+    {
+        if constexpr (is_collection)
+        {
+            for (auto && [sequence1, sequence2] : sequence_pair_or_pairs)
+                total += algorithm(sequence1, sequence2, edit_distance_cfg).score();
+        }
+        else
+        {
+            auto & [sequence1, sequence2] = sequence_pair_or_pairs;
+            total += algorithm(sequence1, sequence2, edit_distance_cfg).score();
+        }
+    }
```

Comment:
Please swap the `if` and the `for` statements, because `is_collection` is constant and does not need to be decided in each loop cycle.